Library/Formula/opencoarrays.rb

```diff
@@ -0,0 +1,73 @@
+class Opencoarrays < Formula
+  desc "Coarray Fortran runtime library and toolchain, free & open source"
+  homepage "http://opencoarrays.org"
+  url "https://github.com/sourceryinstitute/opencoarrays/releases/download/1.2.0/opencoarrays-1.2.0.tar.gz"
+  sha256 "7d0c3f2dd842b32b2a24ac5fb923bc026d541d9e037205162b39a3d381d9fa83"
+
+  head "https://github.com/sourceryinstitute/opencoarrays.git"
+
+  option "without-test", "Run build time tests (requires properly configured MPICH and firewall settings)"
+
+  depends_on :fortran
+  depends_on "gcc"
+  depends_on :mpi => [:cc, :f90, "with-fortran"]
+  depends_on "cmake" => :build
+
+  fails_with :clang do
+    cause "Only compatible with GCC >= 5.1"
+  end
+
+  fails_with :llvm do
+    cause "Only compatible with GCC >= 5.1"
+  end
+
+  fails_with :gcc => 5.0 do
+    cause "GCC 5.1 or greater is required"
+  end
+
+  def install
+    mkdir "build" do
+      ENV["FC"] = HOMEBREW_PREFIX + "bin/mpif90"
+      ENV["CC"] = HOMEBREW_PREFIX + "bin/mpicc"
```

Comment:
There should be no need to specify the full path. Using `mpif90` and `mpicc` should suffice. Ideally, these won't be needed at all and the build system picks up the environment variables `MPICC` and `MPIFC` that are automatically set by `depends_on :mpi => [:cc, :f90]`.